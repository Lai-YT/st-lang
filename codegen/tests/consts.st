% all compile-time constants, should not produce any code

% globals
const i: int := 0
const b := true
const s: string := "string"

put i
put b
put s
skip

% locals
begin
  const i: int := 0
  const b := true
  const s: string := "string"

  put i
  put b
  put s
end
