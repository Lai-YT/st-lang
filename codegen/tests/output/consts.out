0
true
string

0
true
string
